### PR TITLE
Rename security_policies table to host_data

### PR DESCRIPTION
**Pull request by DomAyre**

### CODE REVIEW

File: doc/operations/code_upgrade.rst [hunk 1]
```diff
@@ -177,7 +177,7 @@ Procedure
 - Members should be use the ``set_constitution`` proposal action to update the constitution scripts.
 - See :ref:`bundle deployment procedure <build_apps/js_app_bundle:Deployment>` to update the JavaScript/TypeScript application.
 
-6. Finally, once the code upgrade process has been successful, the old code version (i.e. the code version run by nodes 0, 1 and 2) can be removed using the ``remove_node_code`` or ``remove_security_policy`` proposal actions.
+6. Finally, once the code upgrade process has been successful, the old code version (i.e. the code version run by nodes 0, 1 and 2) can be removed using the ``remove_node_code`` or ``remove_host_data`` proposal actions.
```
Comment (achamayou) [inline doc/operations/code_upgrade.rst:180]: remove_snp_host_data I think, we want to namespace this.

File: doc/audit/builtin_maps.rst [hunk 2]
```diff
@@ -143,14 +143,14 @@ Versions of the code allowed to join the current network on :doc:`SGX <../operat
 
 .. note:: On versions ``3.0.0-dev1`` and below, value was an enum with one possible value ``ALLOWED_TO_JOIN``
 
-``nodes.security_policies``
+``nodes.host_data``
```
Comment (achamayou) [inline doc/audit/builtin_maps.rst:146]: nodes.snp.host_data, or nodes.snp_host_data (I would prefer the former, just to group the snp tables)

File: doc/operations/code_upgrade.rst (hunk 1 above)
Comment (DomAyre) [reply doc/operations/code_upgrade.rst]: 5143a02dcb85d2aacabb5c1fd88e756aed95166a

File: doc/audit/builtin_maps.rst (hunk 2 above)
Comment (DomAyre) [reply doc/audit/builtin_maps.rst:146]: 5143a02dcb85d2aacabb5c1fd88e756aed95166a

File: doc/audit/builtin_maps.rst [hunk 3]
```diff
@@ -143,14 +143,14 @@ Versions of the code allowed to join the current network on :doc:`SGX <../operat
...
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (jumaffre) [inline doc/audit/builtin_maps.rst:147]:
```suggestion
``nodes.host_data``
~~~~~~~~~~~~~~~~~~~
```

File: doc/audit/builtin_maps.rst [hunk 4]
```diff
@@ -143,14 +143,14 @@ Versions of the code allowed to join the current network on :doc:`SGX <../operat
...
 
-Security policies that nodes can run and join the current network on :doc:`SNP <../operations/platforms/snp>`.
+Host data for nodes that can join the current network on :doc:`SNP <../operations/platforms/snp>`.
```
Comment (jumaffre) [inline doc/audit/builtin_maps.rst:149]:
```suggestion
Attestation report host data field for nodes that are allowed to join the current network on :doc:`SNP <../operations/platforms/snp>`.
```

File: include/ccf/service/tables/host_data.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+#include "ccf/crypto/sha256_hash.h"
+#include "ccf/service/map.h"
+
+using HostData = crypto::Sha256Hash;
+using HostDataMetadata = std::string;
+
+namespace ccf
+{
+  using HostDataMap = ServiceMap<HostData, HostDataMetadata>;
```
Comment (jumaffre) [inline include/ccf/service/tables/host_data.h:13]:
```suggestion
  using SnpHostDataMap = ServiceMap<HostData, HostDataMetadata>;
```

File: src/endpoints/common_endpoint_registry.cpp [hunk 6]
```diff
@@ -170,33 +170,33 @@ namespace ccf
...
+          if (!key.has_value() || key.value() == host_data_str)
+            out.host_data.push_back({host_data_str, security_policy});
           return true;
         });
 
       return make_success(out);
     };
     make_read_only_endpoint(
-      "/security_policy",
+      "/host_data",
```
Comment (jumaffre) [inline src/endpoints/common_endpoint_registry.cpp:195]: /snp/host_data

File: src/node/rpc/node_call_types.h [hunk 7]
```diff
@@ -60,7 +60,7 @@ namespace ccf
       QuoteInfo quote_info;
       crypto::Pem public_encryption_key;
       CodeDigest code_digest;
-      std::optional<RawPolicy> security_policy;
+      std::optional<HostDataMetadata> security_policy;
```
Comment (jumaffre) [inline src/node/rpc/node_call_types.h:63]:
```suggestion
      std::optional<HostDataMetadata> host_data_metadata;
```

**Review by jumaffre — APPROVED** [review]

File: src/endpoints/common_endpoint_registry.cpp (hunk 6 above)
Comment (DomAyre) [reply src/endpoints/common_endpoint_registry.cpp]: https://github.com/microsoft/CCF/pull/4374/commits/9924dd14b2000b2452174e36351f90825b13a485

File: src/node/rpc/node_call_types.h (hunk 7 above)
Comment (DomAyre) [reply src/node/rpc/node_call_types.h:63]: I've reverted this, this is where we pass the raw security policy in through the env variable, so here it is still explicitly a security policy

File: samples/constitutions/default/actions.js [hunk 8]
```diff
@@ -959,55 +959,48 @@ const actions = new Map([
...
-          args.security_policy_digest,
-          "string",
-          "security_policy_digest"
-        );
+        checkType(args.host_data, "string", "host_data");
       },
       function (args) {
-        const digest = ccf.strToBuf(args.security_policy_digest);
-        ccf.kv["public:ccf.gov.nodes.security_policies"].delete(digest);
+        const host_data = ccf.strToBuf(args.host_data);
```
Comment (MahatiC) [inline samples/constitutions/default/actions.js:1002]: Should this be called `host_data_digest` instead to be clear that the argument is a digest?

File: src/node/quote.cpp [hunk 9]
```diff
@@ -57,16 +57,16 @@ namespace ccf
     return unique_id;
   }
 
-  std::optional<DigestedPolicy> AttestationProvider::get_security_policy_digest(
+  std::optional<HostData> AttestationProvider::get_host_data(
```
Comment (MahatiC) [inline src/node/quote.cpp:60]: here as well

File: tests/infra/is_snp.py [hunk 10]
```diff
@@ -5,7 +5,7 @@
 
 IS_SNP = os.path.exists("/dev/sev")
 # 4070: Get these from the container
-DEFAULT_SNP_SECURITY_POLICY_DIGEST = (
+DEFAULT_SNP_HOST_DATA = (
```
Comment (MahatiC) [inline tests/infra/is_snp.py:8]: here too

File: samples/constitutions/default/actions.js (hunk 8 above)
Comment (DomAyre) [reply samples/constitutions/default/actions.js:1002]: Host data is an abstract concept on SNP see [here](https://www.amd.com/system/files/TechDocs/56860.pdf#89) and then in confidential containers specifically it happens to be a security policy digest.

Comment (MahatiC) [reply samples/constitutions/default/actions.js:1002]: Ah so this does not actually store the digest.

**Review by achamayou — APPROVED** [review]

**Pull request merged**